app/graphql/mutations/profile/associate_systems.rb

```diff
@@ -12,18 +12,13 @@ class AssociateSystems < BaseMutation
 
       def resolve(args = {})
         profile = find_profile(args[:id])
-        hosts = find_hosts(args[:system_ids])
-        profile_hosts = hosts.map do |host|
-          ProfileHost.new(profile_id: profile.id, host_id: host.id)
-        end
-        ProfileHost.import!(profile_hosts)
+        add_inventory_hosts(args[:system_ids])
+        profile.update_hosts(args[:system_ids])
```

Comment:
@bastilian now we update the profile to the exact set of host IDs passed in, not including any existing hosts. If you pass an empty array, the profile will be unassociated with all hosts.